src/misc.ts

```diff
@@ -633,3 +634,20 @@ const getAllRemovedRowsBeforeRowIndex = (rowIndex: number, removedRowsIndexs: nu
 
 export const getDatagridRowDataIndex = (rowIndex: number, removedRowsIndexs: number[]) =>
   rowIndex + getAllRemovedRowsBeforeRowIndex(rowIndex, removedRowsIndexs).length;
+
+export const fuzzySearch = <T>(array: T[], query: string | undefined, keys: string[]) => {
+  if (!query) {
+    return array;
+  }
+
+  const fuse = new Fuse(array, {
+    keys,
+    includeScore: true,
+    threshold: 0.3,
+  });
+
+  return fuse
+    .search(query.toLocaleLowerCase())
+    .filter(({ score }) => (query ? score !== 1 : 0))
```

Comment:
question: do you have to check `query` if you have a condition checking this above?
Previous solution use a threshold, maybe we should also consider using it.